### PR TITLE
use volume ID for creating PVC from another PVC

### PR DESCRIPTION
**Pull request by Madhu-1**
**What type of PR is this?**

> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change

/kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

current PVC request is sending the PV name while creating a volume from a volume, as per the CSI spec we need to send the volume_id for creation of PVC from another PVC.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
This PR send the volume_id instead of  PVC Name to the csi-driver in volumeCreate when datasource  is PVC
```

### CODE REVIEW

**Comment by msau42** [comment]
/ok-to-test

Please also add a release note

File: pkg/controller/controller.go [hunk 1]
```diff
@@ -671,9 +671,18 @@ func (p *csiProvisioner) getPVCSource(options controller.ProvisionOptions) (*csi
...
+	}
+
+	if sourcePV.Spec.CSI == nil {
+		return nil, fmt.Errorf("error getting volume source from persistantVolumeClaim:persistanceVolume %s:%s", sourcePVC.Name, sourcePVC.Spec.VolumeName)
+	}
+
 	volumeSource := csi.VolumeContentSource_Volume{
 		Volume: &csi.VolumeContentSource_VolumeSource{
-			VolumeId: sourcePVC.Spec.VolumeName,
+			VolumeId: sourcePV.Spec.CSI.VolumeHandle,
```
Comment (msau42) [inline pkg/controller/controller.go]: Do we have a test for this?

Comment (Madhu-1) [reply pkg/controller/controller.go]: updated test case

**Comment by Madhu-1** [comment]
> /ok-to-test
> 
> Please also add a release note

@msau42 sorry where I have to add a release note

**Comment by msau42** [comment]
Please add the release note in the original PR comment, where the blank box currently is

File: pkg/controller/controller.go (hunk 1 above)
Comment (msau42) [reply pkg/controller/controller.go]: Thanks, can the unit test also validate the pv spec has the correct volumeid?

Comment (Madhu-1) [reply pkg/controller/controller.go]: @msau42 this is already done here https://github.com/kubernetes-csi/external-provisioner/blob/master/pkg/controller/controller_test.go#L2195

**Comment by Madhu-1** [comment]
@msau42 added a release note PTAL

**Comment by msau42** [comment]
/approve
Will let @j-griffith lgtm

File: pkg/controller/controller.go [hunk 2]
```diff
@@ -671,9 +671,18 @@ func (p *csiProvisioner) getPVCSource(options controller.ProvisionOptions) (*csi
 		return nil, fmt.Errorf("error, new PVC request must be greater than or equal in size to the specified PVC data source, requested %v but source is %v", requestedSize, sourcePVC.Spec.Size())
 	}
 
+	sourcePV, err := p.client.CoreV1().PersistentVolumes().Get(sourcePVC.Spec.VolumeName, metav1.GetOptions{})
```
Comment (j-griffith) [inline pkg/controller/controller.go]: The `PVC.Spec.VolumeName` field is "optional", would it be more reliable to just use the `Volume` field of the PVC directly?  Then you can do something like `volumeHandleToId` to set the `VolumeId` field below.

Comment (Madhu-1) [reply pkg/controller/controller.go]: do we have `volume` field in PVC spec? 

The approach I followed here is to get the PV name from PVC and then fetch volumeID from PV spec. do we have any other way of getting the volume name from the PVC?

Comment (j-griffith) [reply pkg/controller/controller.go]: Actually you're correct here, Spec.VolumeName is the only handle; and the CO should set this when the claim is bound so this should be fine.  You might want to add a check that the Spec.VolumeName value is not nil before issuing the Get call for the PV though.

**Comment by Madhu-1** [comment]
@j-griffith done PTAL

**Pull request merged**